### PR TITLE
feat: github app integration

### PR DESCRIPTION
**Pull request by thisisnithin**
## Motivation and Context

Github app integration for subgraph schema checks. 

https://github.com/wundergraph/cosmo/assets/44922285/7b875c90-90aa-4e19-b45c-c38b375bf473

Completes ENG-2917
Fixes ENG-4185

- [ ] Tests or benchmark included
- [ ] Documentation is changed or added on [https://app.gitbook.com/](https://app.gitbook.com/)
- [x] PR title must follow [conventional-commit-standard](https://github.com/wundergraph/wundergraph/blob/main/CONTRIBUTING.md#conventional-commit-standard)

### CODE REVIEW

**Comment by StarpTech** [comment]
Well done! :clap:

File: controlplane/src/core/bufservices/PlatformService.ts [hunk 1]
```diff
@@ -485,6 +502,20 @@ export default function (opts: RouterOptions): Partial<ServiceImpl<typeof Platfo
           }
         }
 
+        if (req.gitInfo && opts.githubApp) {
+          const githubRepo = new GitHubRepository(opts.db, opts.githubApp);
+          githubRepo.createCommitCheck({
```
Comment (StarpTech) [inline controlplane/src/core/bufservices/PlatformService.ts:507]: Missing `.catch` otherwise it leads to `unhandledPromiseRejection`

**Review by StarpTech — CHANGES_REQUESTED** [review]

File: controlplane/src/core/bufservices/PlatformService.ts (hunk 1 above)
Comment (thisisnithin) [reply controlplane/src/core/bufservices/PlatformService.ts]: Added await

File: controlplane/migrations/0043_foamy_adam_destine.sql [hunk 2]
```diff
@@ -0,0 +1,19 @@
+DO $$ BEGIN
+ CREATE TYPE "git_installation_type" AS ENUM('PERSONAL', 'ORGANIZATION');
+EXCEPTION
+ WHEN duplicate_object THEN null;
+END $$;
+--> statement-breakpoint
+CREATE TABLE IF NOT EXISTS "git_installations" (
```
Comment (JivusAyrus) [inline controlplane/migrations/0043_foamy_adam_destine.sql:7]: dont we need the organization id?

**Review by JivusAyrus — CHANGES_REQUESTED** [review]

File: controlplane/migrations/0043_foamy_adam_destine.sql (hunk 2 above)
Comment (thisisnithin) [reply controlplane/migrations/0043_foamy_adam_destine.sql:7]: We dont scope this to organization.

Comment (StarpTech) [reply controlplane/migrations/0043_foamy_adam_destine.sql:7]: Should we?

**Comment by StarpTech** [comment]
I'm going to test the helm chart on Monday. After that, we can merge it.

**Review by StarpTech — APPROVED** [review]
LGTM

**Pull request merged**